### DISCUSSION TITLE
Inlay hints for `where` bindings?

### DISCUSSION THREAD

**Discussion by gctypo** [mrcjkb/haskell-tools.nvim#271] [Q&A] [answered]
Is it possible to get inlay hints for `where` bindings? I'm very new to haskell but being able to see what type a `where` binding evaluates to would help me avoid a lot of very dumb type errors.

**Comment by mrcjkb** [comment, marked as answer]
Hey 👋 

haskell-language-server doesn't support inlay hints yet (what you see at the top level are code lenses).

See https://github.com/haskell/haskell-language-server/issues/2938

It's good practice to use the [`ScopedTypeVariables`](https://wiki.haskell.org/Scoped_type_variables) extension and add type signatures to let/where bindings.

**Reply by gctypo** [reply]
Thanks so much, never knew about `ScopedTypeVariables`.
